examples/networkgraph.cpp

```diff
@@ -0,0 +1,909 @@
+/*
+  Copyright 2025 Equinor ASA.
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <iostream>
+#include <getopt.h>
+#include <sstream>
+
+#include <opm/io/eclipse/ERst.hpp>
+
+#include <opm/input/eclipse/Deck/Deck.hpp>
+
+#include <opm/input/eclipse/Parser/ErrorGuard.hpp>
+#include <opm/input/eclipse/Parser/InputErrorAction.hpp>
+#include <opm/input/eclipse/Parser/Parser.hpp>
+#include <opm/input/eclipse/Parser/ParseContext.hpp>
+
+
+class Node
+{
+public:
+    Node(const std::string& name);
+
+    std::string name() const { return m_name; }
+    void set_outlet(std::shared_ptr<Node> outlet) { m_outlet = outlet; };
+    std::shared_ptr<Node> get_outlet() { return m_outlet; };
+
+    void add_inlet_node(std::shared_ptr<Node> n);
+
+    void set_vfp(int vfp){m_vfp = vfp;};
```

Comment:
Lots of inconsistent formatting in the file, such as missing whitespace here. I suggest running clang-format on the file for a quick and easy fix. (There is a .clang-format file in the root of the repo that should apply automatically, I hope, but you can tweak it if you are unhappy with the results).